node/service/src/service.rs

```diff
@@ -423,18 +436,41 @@ where
 
 /// Build the import queue for the parachain runtime.
 #[allow(clippy::type_complexity)]
-pub fn parachain_build_import_queue(
-	client: Arc<TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<TemplateRuntimeExecutor>>>,
+pub fn parachain_build_import_queue<RuntimeApi, Executor>(
+	client: Arc<TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>>,
 	config: &Configuration,
 	telemetry: Option<TelemetryHandle>,
 	task_manager: &TaskManager,
 ) -> Result<
 	sc_consensus::DefaultImportQueue<
 		Block,
-		TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<TemplateRuntimeExecutor>>,
+		TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>,
 	>,
 	sc_service::Error,
-> {
+>
+where
+	RuntimeApi: ConstructRuntimeApi<Block, TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>>
+		+ Send
+		+ Sync
+		+ 'static,
+	RuntimeApi::RuntimeApi: sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>
+		+ sp_api::Metadata<Block>
+		+ sp_session::SessionKeys<Block>
+		+ sp_api::ApiExt<
+			Block,
+			StateBackend = sc_client_api::StateBackendFor<TFullBackend<Block>, Block>,
+		> + sp_offchain::OffchainWorkerApi<Block>
+		+ sp_consensus_aura::AuraApi<Block, AuraId>
+		+ sp_block_builder::BlockBuilder<Block>
+		+ cumulus_primitives_core::CollectCollationInfo<Block>
+		+ pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance>
+		+ substrate_frame_rpc_system::AccountNonceApi<Block, AccountId, Nonce>
+		+ pallet_messages_runtime_api::MessagesApi<Block, AccountId, BlockNumber>
+		+ pallet_schemas_runtime_api::SchemasRuntimeApi<Block>
+		+ pallet_msa_runtime_api::MsaApi<Block, AccountId, BlockNumber>,
```

Comment:
Are we going to need to add a line here whenever we create or want to use a new pallet? Which pallets must be listed? Is there a default import queue that gets added before this? I ask because these aren't all the pallets in the dependencies list. Let's be sure to document this somewhere.

Reply:
Good catch we dont need frequency related pallets here but only needs to be extended in start_node_impl and start_parachain_node.  I have removed from here

About documentation this is fairly known but I will create a chore to discuss about how we let developers know where to add their runtime api when creating a new pallet with rpc

Reply:
I have added a chore https://github.com/LibertyDSNP/frequency/issues/295

Basically so far when we have added runtime api we figured out by ourselves but will be a good time to document some of these for future devs

Reply:
sounds good, thanks